File: iosApp/iosApp/DesignSystem/Theme.swift
import SwiftUI

extension Color {
    init(hex: String) {
        let s = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
        var v: UInt64 = 0
        Scanner(string: s).scanHexInt64(&v)
        let r = Double((v >> 16) & 0xFF) / 255
        let g = Double((v >> 8) & 0xFF) / 255
        let b = Double(v & 0xFF) / 255
        self.init(.sRGB, red: r, green: g, blue: b, opacity: 1)
    }
}

/// Brand tokens translated from the web app's design system
/// (cream / mauve / rose / gold). The app is light-only — see RootView.
enum Theme {
    static let background = Color(hex: "FAF3EC")
    static let surface    = Color(hex: "FFFDF9")
    static let primary    = Color(hex: "6B3FA0")
    static let secondary  = Color(hex: "D4849A")
    static let accent     = Color(hex: "E8C97E")
    static let ink        = Color(hex: "2D2D2D")
    static let inkSoft    = Color(hex: "6B6B6B")
    /// Gold accent tuned for TEXT/icons: deep goldenrod, readable on cream.
    /// Use this for accent text, not `accent`, which is too pale to read.
    static let accentText = Color(hex: "8A6A1C")

    static let phaseMenstrual  = Color(hex: "D96B6B")
    static let phaseFollicular = Color(hex: "5CB37E")
    static let phaseOvulatory  = Color(hex: "E8B84D")
    static let phaseLuteal     = Color(hex: "8A5FC2")

    static var backgroundGradient: LinearGradient {
        LinearGradient(
            colors: [background, Color(hex: "F3E7DA")],
            startPoint: .top, endPoint: .bottom
        )
    }
}

extension Font {
    /// Cycluna display serif — Georgia, matching the approved onboarding design.
    /// `relativeTo` makes it scale with the user's Dynamic Type setting (accessibility);
    /// callers additionally scale `size` by screen width for cross-device proportion.
    static func cyclunaSerif(_ size: CGFloat) -> Font {
        .custom("Georgia", size: size, relativeTo: .largeTitle)
    }
}

extension View {
    /// Solid cream card with subtle border + soft mauve shadow (web `Card shadow-soft`).
    func cyclunaCard(padding: CGFloat = 20, radius: CGFloat = 20) -> some View {
        self
            .padding(padding)
            .frame(maxWidth: .infinity)
            .background(Theme.surface, in: RoundedRectangle(cornerRadius: radius))
            .overlay(RoundedRectangle(cornerRadius: radius).stroke(Theme.inkSoft.opacity(0.12)))
            .shadow(color: Theme.primary.opacity(0.10), radius: 16, x: 0, y: 6)
    }
}

File: androidApp/src/main/kotlin/app/cycluna/android/designsystem/Theme.kt
package app.cycluna.android.designsystem

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Typography
import androidx.compose.material3.lightColorScheme
import androidx.compose.runtime.Composable
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.shadow
import androidx.compose.ui.graphics.Brush
import androidx.compose.ui.graphics.Color
import androidx.compose.ui.text.TextStyle
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.unit.Dp
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp

/**
 * Brand tokens, shared with the iOS app's `Theme.swift` value for value.
 *
 * The app is light-only: no dynamic colour, no dark scheme, and `isSystemInDarkTheme()` is
 * never read. The cream/mauve palette is the brand, and the manifest theme sets
 * `forceDarkAllowed=false` so the system cannot invert it either.
 */
object Theme {
    val background = Color(0xFFFAF3EC)
    val surface = Color(0xFFFFFDF9)
    val primary = Color(0xFF6B3FA0)
    val secondary = Color(0xFFD4849A)
    val accent = Color(0xFFE8C97E)

    /** Gold tuned for TEXT and icons. Use this, not [accent], which is too pale to read. */
    val accentText = Color(0xFF8A6A1C)

    val ink = Color(0xFF2D2D2D)
    val inkSoft = Color(0xFF6B6B6B)

    val phaseMenstrual = Color(0xFFD96B6B)
    val phaseFollicular = Color(0xFF5CB37E)
    val phaseOvulatory = Color(0xFFE8B84D)
    val phaseLuteal = Color(0xFF8A5FC2)

    val backgroundGradient = Brush.verticalGradient(listOf(background, Color(0xFFF3E7DA)))
}

/**
 * The display serif. Georgia on iOS; on Android the platform serif, which is Noto Serif on
 * every device this app supports. Sizes stay in `sp` so the user's font-size setting scales
 * them, which is the Android counterpart of the iOS Dynamic Type relativeTo.
 */
val CyclunaSerif = FontFamily.Serif

private val CyclunaColors = lightColorScheme(
    primary = Theme.primary,
    onPrimary = Color.White,
    secondary = Theme.secondary,
    onSecondary = Color.White,
    background = Theme.background,
    onBackground = Theme.ink,
    surface = Theme.surface,
    onSurface = Theme.ink,
    surfaceVariant = Theme.surface,
    onSurfaceVariant = Theme.inkSoft,
    error = Theme.phaseMenstrual,
)

@Composable
fun CyclunaTheme(content: @Composable () -> Unit) {
    MaterialTheme(
        colorScheme = CyclunaColors,
        typography = Typography(),
        content = content,
    )
}

/** Display serif at a given size — the counterpart of `Font.cyclunaSerif(_:)`. */
fun serif(size: Int): TextStyle = TextStyle(fontFamily = CyclunaSerif, fontSize = size.sp)

/**
 * Solid cream card with a hairline border and a soft mauve shadow (web `Card shadow-soft`).
 * Order matters: the shadow has to be drawn before the background, or it lands on top of it.
 */
fun Modifier.cyclunaCard(padding: Dp = 20.dp, radius: Dp = 20.dp): Modifier {
    val shape = RoundedCornerShape(radius)
    return this
        .fillMaxWidth()
        .shadow(elevation = 8.dp, shape = shape, ambientColor = Theme.primary, spotColor = Theme.primary)
        .background(Theme.surface, shape)
        .border(1.dp, Theme.inkSoft.copy(alpha = 0.12f), shape)
        .padding(padding)
}
